plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Transform/ForceCase.hs

```diff
@@ -0,0 +1,203 @@
+{-# LANGUAGE BangPatterns   #-}
+{-# LANGUAGE LambdaCase     #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+{- | We want to push 'Force' applied at the top of 'Case' terms into the 'Case' branches.
+This transformation is unsound in general, but it can be done for the particular case when
+the term matches:
+
+@
+force
+    ( case
+        (constr i [v1 ... vn])
```

Comment:
The problem is that this case is not interesting: if we have something like this we'll just remove it with `CaseReduce`